proxy/src/coco/signer.rs

```diff
@@ -0,0 +1,130 @@
+//! Signing and key handling.
+
+use std::convert::Infallible;
+
+use librad::keys;
+use librad::paths;
+use librad::signer;
+use radicle_keystore::crypto;
+use radicle_keystore::file;
+use radicle_keystore::sign::{self, Signer as _};
+use radicle_keystore::{FileStorage, Keystore, SecretKeyExt};
+
+pub use librad::keys::SignError;
+pub use radicle_keystore::pinentry::SecUtf8;
+
+/// Filename of the key on disk.
+const KEY: &str = "librad.key";
+
+/// Blanket trait to use as our generic [`signer::Signer`].
+pub trait Signer: Clone + keys::AsPKCS8 + signer::Signer {}
+
+impl<T: Clone + keys::AsPKCS8 + signer::Signer> Signer for T {}
+
+/// Utility trait to manage state for an exisitng [`Signer`].
+pub trait Reset: Signer {
+    /// Reset the [`Signer`] state under the given path.
+    ///
+    /// # Errors
+    ///
+    /// * if the write of the newly generated key fails
+    fn reset(&mut self, paths: &paths::Paths, passphrase: SecUtf8) -> Result<(), Error>;
+}
+
+/// Synonym for an error when interacting with a store for [`librad::keys`].
+pub type Error = file::Error<crypto::SecretBoxError<Infallible>, keys::IntoSecretKeyError>;
+
+/// Parameterised [`FileStorage`] used in Store implementation.
+type Storage = FileStorage<
+    crypto::Pwhash<SecUtf8>,
```

Comment:
I suppose this is still temporary? Because, keeping the passphrase in memory is as good as keeping the secret key itself in memory.

Reply:
See https://github.com/radicle-dev/radicle-upstream/pull/768#issuecomment-676497331